README: Update project info and technical aims/support (#340)
CliMA/ClimaCore.jl: README.md

[zenodo-img]: https://zenodo.org/badge/356355994.svg
[zenodo-url]: https://zenodo.org/badge/latestdoi/356355994

> The Climate Modelling Alliance ([CliMA](https://clima.caltech.edu/)) project is developing a full Earth System Model (ESM), entirely in the high-level, dynamic [Julia](https://julialang.org/) language. The main goal of the project is to build an ESM that automatically learns from diverse data sources to produce accurate climate predictions with quantified uncertainties. The CliMA model has been developed to target both CPU and GPU architectures, using a common codebase. ClimaCore.jl constitutes the dynamical core (_dycore_), providing discretization tools to solve the governing equations describing the different components of the ESM.

Reviewer: ```suggestion
The Climate Modelling Alliance ([CliMA](https://clima.caltech.edu/)) is developing a new Earth System Model (ESM), entirely written in the [Julia](https://julialang.org/) language. The main goal of the project is to build an ESM that automatically learns from diverse data sources to produce accurate climate predictions with quantified uncertainties. The CliMA model targets both CPU and GPU architectures, using a common codebase. ClimaCore.jl constitutes the dynamical core (_dycore_) of the atmosphere and land models, providing discretization tools to solve the governing equations of the ESM component models.
```

> ClimaCore.jl's high-level application programming interface (API) facilitates modularity and composition of differential operators and the definition of flexible discretizations. This, in turn, is coupled with the low-level API that supports different data layouts, specialized implementations and flexible models for threading, to better face high-performance optimization, data storage, and scalability challenges on modern HPC ecosystems.

Reviewer: ```suggestion
ClimaCore.jl's high-level application programming interface (API) facilitates modularity and composition of differential operators and the definition of flexible discretizations. This, in turn, is coupled with low-level APIs that support different data layouts, specialized implementations, and flexible models for threading, to better face high-performance optimization, data storage, and scalability challenges on modern HPC architectures.
```

## Technical aims and current support

> * Support both Large Eddy Simulation (LES) and General Circulation Model (GCM) configurations.

Reviewer: ```suggestion
* Support both large-eddy simulation (LES) and general circulation model (GCM) configurations for the atmosphere.
```

> * Limit number of languages (ideally one: currently, Julia).

Reviewer: ```suggestion
```

> * A suite of tools for constructing spatial compatible discretizations.

Reviewer: ```suggestion
* A suite of tools for constructing space discretizations.
```

* Horizontal spectral elements:

>     - Supports both Continuous (SEM) and Discontinuous Galerkin (DG-SEM) discretizations.

Reviewer: ```suggestion
    - Supports both continuous Galerkin (CG) and discontinuous Galerkin (DG) spectral element discretizations.
```

> * Flexible choice of vertical discretization (currently staggered Finite Differences, potentially Finite Volumes)

Reviewer: ```suggestion
* Flexible choice of vertical discretization (currently staggered finite differences)
```

> * Support for different geometric vector types and coordinate systems: Covariant and Contravariant vectors for curvilinear, non-orthogonal systems; Cartesian vectors for Eucledean spaces; and UVVectors for lat-long grids.

Reviewer: ```suggestion
* Support for different geometries (Cartesian, spherical), with governing equations discretized in terms of covariant  vectors for curvilinear, non-orthogonal systems and Cartesian vectors for Euclidean spaces.
```

* `Field` abstraction:
    - Scalar, vector or struct-valued
    - Stores values, geometry, and mesh information
    - Flexible memory layouts: Array-of-Structs (AoS), Struct-of-Arrays (SoA),Array-of-Struct-of-Arrays (AoSoA)
    - Useful overloads: `sum` (integral), `norm`, etc.
    - Compatible with [`DifferentialEquations.jl`](https://diffeq.sciml.ai/stable/) time steppers.

> * Composable operators via broadcasting: Apply a function element-wise to an array; Scalar values are broadcast over arrays Fusion of multiple operations; Can be specialized for custom functions or argument types (e.g. `CuArray` compiles and applies a custom CUDA kernel).

Reviewer: ```suggestion
* Composable operators via broadcasting: apply a function element-wise to an array; scalar values are broadcast over arrays 
* Fusion of multiple operations; can be specialized for custom functions or argument types (e.g. `CuArray` compiles and applies a custom CUDA kernel).
```

> * Operators (`grad`, `div`, `interpolate`) are “pseudo-functions”: Act like functions when broadcasted over a `Field`; Fuse operators and function calls.

Reviewer: ```suggestion
* Operators (`grad`, `div`, `interpolate`) are “pseudo-functions”: Act like functions when broadcasted over a `Field`; fuse operators and function calls.
```

> * Add element node sizes dimensions to type domain

Reviewer: ```suggestion
* Add element node size dimensions to type domain
```

Reviewer 2: I think this is mostly an implementation detail

Author: Ok, we can remove it from here

Author: but I think its nested bullet points are useful and I thought the sentence served to give context

>     - i.e. specialize on polynomial degree

Reviewer: ```suggestion
    - i.e., specialize on polynomial degree
```

Reviewer 2: also kind of an implementation detail, not sure if it is worth highlighting
